Clean/Utils/Tactics.lean

```diff
@@ -54,4 +56,192 @@ partial def SpecializeAuto (e_term : Lean.Term): Lean.Elab.Tactic.TacticM Unit :
+        | .const name _ =>
+          try
+            let instType ← mkAppM ``ProvableStruct #[.const name []]
+            if let .some _ ← trySynthInstance instType then
+              result := result ++ [fvarId]
+          catch _ => continue
+        | _ => continue
+    
+    -- Also search for projections in the goal
+    let goal ← getMainGoal
+    let target ← goal.getType
+    let varsInGoal ← findStructVars target
+    -- Filter to only include those with ProvableStruct instances
+    for fvarId in varsInGoal do
+      let type ← inferType (.fvar fvarId)
+      let type' ← withTransparency .reducible (whnf type)
+      let typeCtor := type'.getAppFn
+      match typeCtor with
+      | .const name _ =>
+        try
+          let instType ← mkAppM ``ProvableStruct #[.const name []]
+          if let .some _ ← trySynthInstance instType then
+            result := result ++ [fvarId]
+        catch _ => continue
+      | _ => continue
+    
+    -- Remove duplicates and return
+    return result.eraseDups
+
+/--
+  Decompose all variables with ProvableStruct instances in the context
+-/
+def decomposeProvableStruct : Lean.Elab.Tactic.TacticM Unit := do
+  withMainContext do
+    let goal ← getMainGoal
+    
+    -- Find all variables with ProvableStruct instances that appear in projections
+    let fvarIds ← findProvableStructVars
+    
+    if fvarIds.isEmpty then
+      throwError "No variables with ProvableStruct instances found in the context or goal"
+    
+    -- Apply cases on each variable
+    let mut currentGoal := goal
+    let mut decomposed := false
+    
+    for fvarId in fvarIds do
+      let localDecl ← fvarId.getDecl
+      let userName := localDecl.userName
+      
+      try
+        -- Use cases tactic on the variable
+        let casesResult ← currentGoal.cases fvarId
+        
+        -- Get the new goal from the cases result
+        match casesResult.toList with
+        | [subgoal] => 
+          currentGoal := subgoal.mvarId
+          decomposed := true
+        | _ => 
+          throwError s!"Unexpected result from cases on {userName}"
+      catch _ =>
+        -- Skip variables that can't be decomposed
+        -- This can happen with type synonyms or other non-destructurable types
+        continue
+    
+    if not decomposed then
+      throwError "Failed to decompose any variables with ProvableStruct instances"
+    
+    replaceMainGoal [currentGoal]
+
+
+/--
+  Automatically decompose ALL variables with ProvableStruct instances that either:
+  1. Are in the context as variables with ProvableStruct instances
+  2. Appear in field projections in hypotheses (e.g., h : input.x = 5)
+  3. Appear in field projections in the goal
+  
+  Example:
+  ```lean
+  theorem example_theorem (input : Inputs (F p)) (h : input.x = 5) : 
+    input.y + input.z = someValue := by
+    decompose_provable_struct  -- This will destruct `input` (found via projections)
+    -- Now we have x, y, z in context with h : x = 5
```

Comment:
The new variables are not explicitly named. The user would need to call `rename_i` if they need names.

Reply:
Updated the document.